core/backend.go

```diff
@@ -30,20 +33,24 @@ type MessageConstructor interface {
 
 // Verifier defines the verifier interface
 type Verifier interface {
-	// IsValidBlock checks if the proposed block is child of parent
+	// IsValidBlock checks if the proposed block is valid and child of the latest block in local chain
```

Comment:
Why are we assuming that the backed will perform this validations? WDYT of just `IsValidBlock checks if the proposed block is valid`

Reply:
Sorry, I'm afraid I don't understand what you said. What is your concern? This method needs to check the validity of block and that it's the next of the latest block.

Reply:
We're assuming that the "Backend" will perform a specific validation (checking that the block is child of the latest block in chain) but if this is an agnostic library, the backend does not necessarily needs to check that, maybe it will check another thing, I know that this is highly improbable but who knows what backend implementations will this be used for? something esoteric maybe... 😄